src/files.rs

```diff
@@ -137,25 +137,35 @@ pub fn guess_content(bytes: &[u8]) -> ProbableFileKind {
         return ProbableFileKind::Text(valid_utf8_string.to_string());
     }
 
-    // If the bytes are is entirely valid UTF-16, treat them as a
-    // string.
-    let u16_values = u16_from_bytes(bytes);
-    if let Ok(valid_utf16_string) = String::from_utf16(&u16_values) {
-        return ProbableFileKind::Text(valid_utf16_string);
-    }
-
     // Only consider the first 1,000 bytes, as tree_magic_mini
     // considers the entire file, which is very slow on large files.
     let mut magic_bytes = bytes;
     if magic_bytes.len() > 1000 {
         magic_bytes = &magic_bytes[..1000];
     }
 
+    let mime = tree_magic_mini::from_u8(magic_bytes);
+    info!("MIME type detected: {}", mime);
+
+    // This is a dirty hack to handle cases where files are obviously
+    // binary but can be still converted to UTF16 string.
+    // See test_gzip_is_binary
+    match mime {
+        "application/octet-stream" => return ProbableFileKind::Binary,
```

Comment:
This isn't quite right.

```
$ RUST_LOG=info cargo run -- sample_files/utf16_before.py sample_files/utf16_after.py
 2023-02-01T23:39:05.289Z INFO  difft::options > CLI arguments: ["sample_files/utf16_before.py", "sample_files/utf16_after.py"]
 2023-02-01T23:39:05.305Z INFO  difft::files   > MIME type detected: application/octet-stream
 2023-02-01T23:39:05.307Z INFO  difft::files   > MIME type detected: application/octet-stream
```

The MIME database treats valid UTF-16 files as octet-stream.

```suggestion
```

Change this check to only consider gzip files, and I'm happy :)

(If you can see better heuristics that would be wonderful, but nothing is obvious to me at least.)

Reply:
Done. You were right that the differences in my local runs and CI was that I didn't have the mime database so tree_magic interpreted pretty much everything to have `application/octet-stream` content type. I installed and setup the db properly and I was able to match the same results as the CI